dev/release/post-06-csharp.sh

```diff
@@ -44,6 +44,16 @@ curl \
 rm -rf ${archive_name}
 tar xf ${tar_gz}
 pushd ${archive_name}/csharp
+
+# Create a dummy .git/ directory to download the source files from GitHub with Source Link.
+mkdir .git && cd $_
```

Comment:
I think this `.git` directory needs to be one directory up, above the `csharp` directory. Basically, it needs to match what a "normal" git repo would look like.